app/services/api_prisoner_checker.rb

```diff
@@ -21,7 +14,28 @@ def error
 
 private
 
+  attr_accessor :noms_id, :date_of_birth
+
   def prisoner_validation
-    @prisoner_validation ||= PrisonerValidation.new(@offender).tap(&:valid?)
+    @prisoner_validation ||= PrisonerValidation.new(offender).tap(&:valid?)
+  end
+
+  def offender
+    @offender ||= fetch_offender? ? fetch_offender : null_offender
+  end
+
+  def fetch_offender?
+    Rails.configuration.nomis_public_prisoner_check_enabled
+  end
+
+  def fetch_offender
+    Nomis::Api.instance.lookup_active_offender(
+      noms_id:       noms_id,
+      date_of_birth: date_of_birth
+    )
+  end
+
+  def null_offender
+    @null_offender ||= Nomis::NullOffender.new(api_call_successful: false)
```

Comment:
No need to memoize since this will get called at most once.

Unrelated, but I wonder if the null offender could be a singleton object.